homeassistant/components/kindhome_solarbeaker/config_flow.py

```diff
@@ -0,0 +1,123 @@
+"""Config flow for Kindhome Solarbeaker integration."""
+import logging
+from typing import Any
+
+import voluptuous as vol
+
+from homeassistant.components.bluetooth import (
+    BluetoothServiceInfoBleak,
+    async_discovered_service_info,
+)
+from homeassistant.config_entries import ConfigFlow
+from homeassistant.const import CONF_ADDRESS
+from homeassistant.data_entry_flow import FlowResult
+
+from .const import DOMAIN
+from .kindhome_solarbeaker_ble import KindhomeSolarbeakerDevice
+from .utils import log
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class KindhomeSolarbeakerConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for kindhome."""
+
+    VERSION = 1
+
+    def _create_config_entry(self, address, name=None):
+        title = f"{name} ({address})"
+        return self.async_create_entry(
+            title=title, data={"address": address, "name": name}
+        )
+
+    def __init__(self) -> None:
+        """Initialize the config flow."""
```

Comment:
There are loads of docstrings that seem unnecessary, like this one. They are actually enforced by _.pre-commit-config.yaml_ in the repo, so I had to add them.